### PR TITLE
Fix tweakreg bug for no source catalogs

### PR DESCRIPTION
**Pull request by hbushouse**
DMS testing recently ran across the following error in the `tweakreg` step when processing a set of (apparently blank) NIRISS images that resulted in no useable sources being in any of the input images:
```
2019-03-25 13:30:17,477 - stpipe.Image3Pipeline.tweakreg - INFO - Step tweakreg running with args (<ModelContainer>,).
2019-03-25 13:30:17,515 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 1 sources in jw87700005001_02102_00001_nis_cal.fits.
2019-03-25 13:30:17,530 - stpipe.Image3Pipeline.tweakreg - WARNING - /Users/bushouse/anaconda3/envs/myjwst/lib/python3.6/site-packages/photutils/detection/findstars.py:977: AstropyUserWarning: Sources were found, but none pass the sharpness and roundness criteria.
  'and roundness criteria.', AstropyUserWarning)

2019-03-25 13:30:17,532 - stpipe.Image3Pipeline.tweakreg - WARNING - No sources found in jw87700005001_02103_00001_nis_cal.fits.
2019-03-25 13:30:17,541 - stpipe.Image3Pipeline.tweakreg - WARNING - /Users/bushouse/anaconda3/envs/myjwst/lib/python3.6/site-packages/photutils/detection/findstars.py:977: AstropyUserWarning: Sources were found, but none pass the sharpness and roundness criteria.
  'and roundness criteria.', AstropyUserWarning)

2019-03-25 13:30:17,543 - stpipe.Image3Pipeline.tweakreg - WARNING - No sources found in jw87700005001_02101_00001_nis_cal.fits.
2019-03-25 13:30:17,557 - stpipe.Image3Pipeline.tweakreg - WARNING - /Users/bushouse/anaconda3/envs/myjwst/lib/python3.6/site-packages/photutils/detection/findstars.py:977: AstropyUserWarning: Sources were found, but none pass the sharpness and roundness criteria.
  'and roundness criteria.', AstropyUserWarning)

2019-03-25 13:30:17,558 - stpipe.Image3Pipeline.tweakreg - WARNING - No sources found in jw87700005001_02104_00001_nis_cal.fits.
----------------------------------------------------------------------
ERROR RUNNING STEP 'Image3Pipeline':
    cannot delete array elements
Traceback (most recent call last):
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.6/site-packages/jwst-0.13.2a0.dev75-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.6/site-packages/jwst-0.13.2a0.dev75-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.6/site-packages/jwst-0.13.2a0.dev75-py3.6-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_image3.py", line 70, in process
    input_models = self.tweakreg(input_models)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.6/site-packages/jwst-0.13.2a0.dev75-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.6/site-packages/jwst-0.13.2a0.dev75-py3.6-macosx-10.7-x86_64.egg/jwst/tweakreg/tweakreg_step.py", line 127, in process
    wgroup = WCSGroupCatalog(wcsimlist, name=wcsimlist[0].name)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.6/site-packages/jwst-0.13.2a0.dev75-py3.6-macosx-10.7-x86_64.egg/jwst/tweakreg/wcsimage.py", line 745, in __init__
    self._catalog = self.create_group_catalog()
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.6/site-packages/jwst-0.13.2a0.dev75-py3.6-macosx-10.7-x86_64.egg/jwst/tweakreg/wcsimage.py", line 889, in create_group_catalog
    del col_catname[0]
ValueError: cannot delete array elements
```

This PR modifies the `create_group_catalog` function to create an empty "cat_name" column in the output dummy source table, rather than creating that column with 1 entry and then trying to delete that entry (which isn't allowed). This allows `tweakreg` to run to completion, with suitable warnings generated saying that no image alignment could be performed.

### CODE REVIEW

**Comment by mcara** [comment]
@hbushouse Thank you for finding a solution! The old code was likely some remnant from a copy/paste from the non-empty catalog branch.

**Comment by hbushouse** [comment]
> @hbushouse Thank you for finding a solution! The old code was likely some remnant from a copy/paste from the non-empty catalog branch.

You're welcome. Gave me the chance to play around a bit with astropy.table routines and see how they work! And it was a pretty simple case (not heavy-duty problems with spherical geometry ...)

**Comment by mcara** [comment]
Yeah, my heart sunk when I saw `spherical_geometry` :)

**Comment by hbushouse** [comment]
@mcara Do you agree with the change? If so, please use the review function to approve, so that this can be merged.

**Comment by mcara** [comment]
Initially I thought this (the creation of a column with a single element and then deleting it) was a copy/paste mistake but now I think I did it intentionally. I was also surprised that I did not test this code (however, I suspect I did at the command line/interpreter). So, now I am trying to install some old astropy to see if the crash is due to the astropy changes. The reason for doing this weird deletion (which I personally do not like) was to create a column of the correct type. This is necessary because an empty catalog can “grow” later (depending on other settings/grouping, etc.) and in that case your code will crash because of the incorrect column type for the file name. With your fix, the column type is wrong for a column that is supposed to hold string names:
```python
>>> col_catname = table.MaskedColumn([], name='cat_name')                                               
>>> col_catname                                                                                         
<MaskedColumn name='cat_name' dtype='float64' length=0>
```

**Comment by mcara** [comment]
@hbushouse, `del col_catname[0]` does not work even in the old `astropy`. I can only guess that I assumed that if `del` works on tables, it will work on columns. Still the issue of the correct data type stands. I found that using boolean indices achieve the same thing as I intended in the oridinal code. Therefor I would suggest that you keep the original column creation code but replace `del col_catname[0]` with
```python
col_catname = col_catname[[False]]
```

File: jwst/tweakreg/wcsimage.py [hunk 1]
```diff
@@ -885,8 +885,7 @@ def create_group_catalog(self):
             else:
                 catname = image.name
 
-            col_catname = table.MaskedColumn([catname], name='cat_name')
```
Comment (mcara) [inline jwst/tweakreg/wcsimage.py]: Keep this line

File: jwst/tweakreg/wcsimage.py [hunk 2]
```diff
@@ -885,8 +885,7 @@ def create_group_catalog(self):
...
-            del col_catname[0]
```
Comment (mcara) [inline jwst/tweakreg/wcsimage.py]: Replace with `col_catname = col_catname[[False]]`

File: jwst/tweakreg/wcsimage.py [hunk 3]
```diff
@@ -885,8 +885,7 @@ def create_group_catalog(self):
...
+            col_catname = table.MaskedColumn([], name='cat_name')
```
Comment (mcara) [inline jwst/tweakreg/wcsimage.py]: This will produce incorrect column type.

File: jwst/tweakreg/wcsimage.py (hunk 2 above)
Comment (hbushouse) [reply jwst/tweakreg/wcsimage.py]: Fixed.

**Pull request merged**